api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -963,32 +965,36 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() throws
 	 * @see VisitService#saveVisit(Visit)
 	 */
 	@Test
-	@Verifies(value = "save visit should persist new encounter", method = "saveVisit(Visit)")
-	public void saveVisit_shouldPersistNewEncounter() throws Exception {
+	@Verifies(value = "save visit should not persist new encounter", method = "saveVisit(Visit)")
+	public void saveVisit_shouldNotPersistNewEncounter() throws Exception {
 		
 		VisitService vs = Context.getVisitService();
 		Visit visit = vs.getVisit(1);
 		
 		Encounter encounter = new Encounter();
 		encounter.setEncounterDatetime(new Date());
+		encounter.setEncounterType(Context.getEncounterService().getEncounterType(1));
 		encounter.setPatient(visit.getPatient());
 		encounter.setLocation(visit.getLocation());
 		visit.addEncounter(encounter);
 		
 		vs.saveVisit(visit);
```

Comment:
Are you sure this test doesn't pass before your changes? I think it passed since you're not flushing the the session before check the encounter id.

I think the test needs to ensure that when the visit is flushed an exception is thrown since the flushed visit would be referencing a transient encounter